model/job-management/src/main/proto/beam_job_api.proto

```diff
@@ -176,3 +179,36 @@ message JobState {
     CANCELLING = 10;
   }
 }
+
+
+// DescribePipelineOptions provides metadata about the options supported by a runner.
+// It will be used by the SDK client to validate the options specified by or
+// list available options to the user.
+// Throws error GRPC_STATUS_UNAVAILABLE if server is down
+message DescribePipelineOptionsRequest {
+}
+
+// Metadata for a pipeline option.
+message PipelineOptionDescriptor {
+  // (Required) The option name.
+  string name = 1;
+
+  // (Required) Type of option.
+  // Matching JSON types, since options are represented as JSON.
+  // "string" | "boolean" | "integer" | "number" | "list" | "map" | "object"
```

Comment:
@mxm this should probably be turned into enum now

Reply:
Can we do this in this PR?

Reply:
yes, would prefer that if there is agreement on the types as such

Reply:
JSON has just `number`, which isn't sufficient for validation. JSON schema addressed that: https://json-schema.org/understanding-json-schema/reference/numeric.html

Reply:
Ah ok, didn't know this was based on JSON schema. Maybe add a link?

Reply:
Changed to enum.